test/core/buffer_cat.cpp

```diff
@@ -66,6 +66,12 @@ class buffer_cat_test : public unit_test::suite
             expect(buffer_size(buffer_cat(
                 sb1.data(), sb2.data())) == 12);
         }
+        for(auto it = bs.begin(); it != bs.end(); ++it)
+        {
+            decltype(bs)::const_iterator copy;
+            copy = it;
+            copy = copy;
```

Comment:
If you're going to test self-assignment, then you should ensure that the result is sane and not just successful. For example, you could `expect (copy == it);` after each of the assignments.